### PR TITLE
Light weight changes from a github first timer

### PR DESCRIPTION
**Pull request by ziuq**
Looking at the code shows the coding standards to be ignored.

use 'v5.10.0';

This not only enforces the prerequisite of a 5.10 or newer Perl install;  but it also disables newer features that were not in 5.10.   This should be in all Perl files. 

I would hope that usage could diminish or eliminate the need to test against various perl versions;
but I cannot vouch for the mechanism's robustness.

rir

### CODE REVIEW

File: lib/LedgerSMB/Scripts/login.pm [hunk 1]
```diff
@@ -18,6 +18,8 @@ This script contains the request handlers for logging in and out of LedgerSMB.
 
 package LedgerSMB::Scripts::login;
 
+use version 'v5.10.0';
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Scripts/login.pm]: Thanks for the contributions. A question regarding this change: In light of the article referenced below, should we discuss this change on the mailing list?

https://www.effectiveperlprogramming.com/2010/07/watch-out-for-side-effects-with-use-version/

File: doc/coding-standard.tex [hunk 2]
```diff
@@ -1,74 +0,0 @@
-\documentclass{article}
```
Comment (ehuelsmann) [inline doc/coding-standard.tex]: Agreed that we can remove this file.

File: doc/intro-coding [hunk 3]
```diff
@@ -0,0 +1,61 @@
+How to contribute code to LedgerSMB.
```
Comment (ehuelsmann) [inline doc/intro-coding]: Wondering if we should add this to the website instead; we have a community guide which explains some (most?) of what I put into my e-mail: https://ledgersmb.org/community-guide

It's probably a good idea to add information like that to the README.md file for people to find...

**Comment by ziuq** [comment]
On Sun, Mar 26, 2017 at 05:10:50AM -0700, Erik Huelsmann wrote:
> ehuelsmann commented on this pull request.

> > @@ -18,6 +18,8 @@ This script contains the request handlers for logging in and out of LedgerSMB.

>  package LedgerSMB::Scripts::login;
>  
> +use version 'v5.10.0';

> Thanks for the contributions. A question regarding this change: In light of the article
> referenced below, should we discuss this change on the mailing list?
> 
> https://www.effectiveperlprogramming.com/2010/07/watch-out-for-side-effects-with-use-version/

It should be the old fashioned

    use 5.010;

or

   use v5.10.0;     # if 5.10 accepts this



My reading of the perldocs made me think the newer vstring version is the preferred replacement,
but I was mistaken about quoting it.  From there, my thinking did not get better.

brian d foy is a good authority.  He usually addresses the fairly adept.  He is only warning of 
language changes that might trip up the unaware.  He also points out that 'no feature' is 
a quick fix if you need it.  He did not bother with the problems of not using 'use 5.nnn_nnn'
because they are traditional.  

I see no good argument to enforcing coding at 5.6/5.8  when enforcing the prerequisite
of 5.10.  We are aiming for big change, so get the users' inconveniences all over with at
once.  I would have suggested 5.24 now rather than force another linux upgrade in two
to four years.

Having browsed through a lot of the code, I did not notice conflicts with new features; so
I don't expect much of a problem.

Rob

**Comment by hasorli** [comment]
If we change the Perl version, I suggest to not change the Perl version to higher than latest Centos 7.3

$perl -v
This is perl 5, version 16, subversion 3 (v5.16.3)

$ perl -V
Summary of my perl5 (revision 5 version 16 subversion 3) configuration:
   
  Platform:
    osname=linux, osvers=2.6.32-573.18.1.el6.x86_64, archname=x86_64-linux-thread-multi
    uname='linux worker1.bsys.centos.org 2.6.32-573.18.1.el6.x86_64 #1 smp tue feb 9 22:46:17 utc 2016 x86_64 x86_64 x86_64 gnulinux '
    config_args='-des -Doptimize=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic -Dccdlflags=-Wl,--enable-new-dtags -Dlddlflags=-shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic -Wl,-z,relro  -DDEBUGGING=-g -Dversion=5.16.3 -Dmyhostname=localhost -Dperladmin=root@localhost -Dcc=gcc -Dcf_by=Red Hat, Inc. -Dprefix=/usr -Dvendorprefix=/usr -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl5 -Dsitearch=/usr/local/lib64/perl5 -Dprivlib=/usr/share/perl5 -Dvendorlib=/usr/share/perl5/vendor_perl -Darchlib=/usr/lib64/perl5 -Dvendorarch=/usr/lib64/perl5/vendor_perl -Darchname=x86_64-linux-thread-multi -Dlibpth=/usr/local/lib64 /lib64 /usr/lib64 -Duseshrplib -Dusethreads -Duseithreads -Dusedtrace=/usr/bin/dtrace -Duselargefiles -Dd_semctl_semun -Di_db -Ui_ndbm -Di_gdbm -Di_shadow -Di_syslog -Dman3ext=3pm -Duseperlio -Dinstallusrbinperl=n -Ubincompat5005 -Uversiononly -Dpager=/usr/bin/less -isr -Dd_gethostent_r_proto -Ud_endhostent_r_proto -Ud_sethostent_r_proto -Ud_endprotoent_r_proto -Ud_setprotoent_r_proto -Ud_endservent_r_proto -Ud_setservent_r_proto -Dscriptdir=/usr/bin -Dusesitecustomize'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.8.5 20150623 (Red Hat 4.8.5-11)', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -fstack-protector'
    libpth=/usr/local/lib64 /lib64 /usr/lib64
    libs=-lresolv -lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lpthread -lc -lgdbm_compat
    perllibs=-lresolv -lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.17'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,--enable-new-dtags -Wl,-rpath,/usr/lib64/perl5/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -Wl,-z,relro '


Characteristics of this binary (from libperl): 
  Compile-time options: HAS_TIMES MULTIPLICITY PERLIO_LAYERS
                        PERL_DONT_CREATE_GVSV PERL_IMPLICIT_CONTEXT
                        PERL_MALLOC_WRAP PERL_PRESERVE_IVUV USE_64_BIT_ALL
                        USE_64_BIT_INT USE_ITHREADS USE_LARGE_FILES
                        USE_LOCALE USE_LOCALE_COLLATE USE_LOCALE_CTYPE
                        USE_LOCALE_NUMERIC USE_PERLIO USE_PERL_ATOF
                        USE_REENTRANT_API USE_SITECUSTOMIZE
  Locally applied patches:
	Fedora Patch1: Removes date check, Fedora/RHEL specific
	Fedora Patch3: support for libdir64
	Fedora Patch4: use libresolv instead of libbind
	Fedora Patch5: USE_MM_LD_RUN_PATH
	Fedora Patch6: Skip hostname tests, due to builders not being network capable
	Fedora Patch7: Dont run one io test due to random builder failures
	Fedora Patch9: Fix find2perl to translate ? glob properly (RT#113054)
	Fedora Patch10: Fix broken atof (RT#109318)
	Fedora Patch13: Clear $@ before "do" I/O error (RT#113730)
	Fedora Patch14: Do not truncate syscall() return value to 32 bits (RT#113980)
	Fedora Patch15: Override the Pod::Simple::parse_file (CPANRT#77530)
	Fedora Patch16: Do not leak with attribute on my variable (RT#114764)
	Fedora Patch17: Allow operator after numeric keyword argument (RT#105924)
	Fedora Patch18: Extend stack in File::Glob::glob, (RT#114984)
	Fedora Patch19: Do not crash when vivifying $|
	Fedora Patch20: Fix misparsing of maketext strings (CVE-2012-6329)
	Fedora Patch21: Add NAME headings to CPAN modules (CPANRT#73396)
	Fedora Patch22: Fix leaking tied hashes (RT#107000) [1]
	Fedora Patch23: Fix leaking tied hashes (RT#107000) [2]
	Fedora Patch24: Fix leaking tied hashes (RT#107000) [3]
	Fedora Patch25: Fix dead lock in PerlIO after fork from thread (RT#106212)
	Fedora Patch26: Make regexp safe in a signal handler (RT#114878)
	Fedora Patch27: Update h2ph(1) documentation (RT#117647)
	Fedora Patch28: Update pod2html(1) documentation (RT#117623)
	Fedora Patch29: Document Math::BigInt::CalcEmu requires Math::BigInt (CPAN RT#85015)
	RHEL Patch30: Use stronger algorithm needed for FIPS in t/op/crypt.t (RT#121591)
	RHEL Patch31: Make *DBM_File desctructors thread-safe (RT#61912)
	RHEL Patch32: Use stronger algorithm needed for FIPS in t/op/taint.t (RT#123338)
	RHEL Patch33: Remove CPU-speed-sensitive test in Benchmark test
	RHEL Patch34: Make File::Glob work with threads again
	RHEL Patch35: Fix CRLF conversion in ASCII FTP upload (CPAN RT#41642)
	RHEL Patch36: Do not leak the temp utf8 copy of namepv (CPAN RT#123786)
	RHEL Patch37: Fix duplicating PerlIO::encoding when spawning threads (RT#31923)
	RHEL Patch38: Backported libraries historically supplied with Perl 4
  Built under linux
  Compiled at Nov  6 2016 01:30:49
  @INC:
    /usr/local/lib64/perl5
    /usr/local/share/perl5
    /usr/lib64/perl5/vendor_perl
    /usr/share/perl5/vendor_perl
    /usr/lib64/perl5
    /usr/share/perl5
    .

**Comment by ziuq** [comment]
On Tue, Mar 28, 2017 at 05:31:33AM -0700, Håvard Sørli wrote:
> If we change the Perl version, I suggest to not change the Perl version to 
> higher than latest Centos 7.3
> 
> $perl -v
> This is perl 5, version 16, subversion 3 (v5.16.3)

It is too late to change the primary prerequisites for lsmb 1.5.
Perl v5.10 was and is a reasonable choice.

Rob

**Comment by ehuelsmann** [comment]
We have discussed increasing the version number minimum requirement and
although I can't remember the exact outcome, I think we were looking at
supporting the lowest Perl version in the latest releases of each of the
biggest distros (ubuntu, debian, centos, fedora) at the time of the release
of the next version (1.6 in this case). By that standard I think we should
indeed be able to upgrade to 5.16 at this point.

Incidentally, when you look at travis ci testson master, we stopped testing
on 5.10 and 5.12.

**Comment by ziuq** [comment]
If we have been considering Perl 5.10 and 5.12 unsupported, then changing up is reasonable.

If we are changing to, or are at, a 5.14 or later Perl.  The concerns brian d foy raised are probably
not germaine.   The comments were from when the behavior was new.   The later Perls are advertised
as able to downshift to lower level versions, so using Perl version eliminates older Perls by death and
newer ones by forcing a compatibility mode.  The concern of running code before use version is worth
following, but that applies to any BEGIN code and should not deter us.

I will explore the soundness of using version.  I know there are many followers of 'modern' Perl, so there
should be a noisy group if the feature is broken.  I will not be fast on this task.

If there are general purpose modules embedded in LSMB,  they might be versioned lower for general
independent use.

File: conf/systemd/ledgersmb-development_plackup.service [hunk 4]
```diff
@@ -34,7 +34,7 @@ ExecStart=/usr/bin/plackup           \
     --access-log $LOG                \
     --server HTTP::Server::PSGI      \
     --env development                \
-    --reload
+    --reload                        \
```
Comment (ehuelsmann) [inline conf/systemd/ledgersmb-development_plackup.service]: The code on master has changed. This change is no longer applicable.

**Comment by ehuelsmann** [comment]
Closing this PR due to multiple issues. @ziuq could you split this PR into:

1. the version string change
2. the removal of the coding standards doc

and add the instructions to the community guide hosted on ledgersmb.org (with a pointer from the toplevel README to that document).

Thanks in advance!

Regards,


Erik.

**Pull request closed without merging**